pkg/gce-cloud-provider/compute/gce.go

```diff
@@ -159,30 +155,40 @@ func readConfig(configPath string) (*ConfigFile, error) {
 	return cfg, nil
 }
 
-func createBetaCloudService(ctx context.Context, vendorVersion string, tokenSource oauth2.TokenSource) (*computebeta.Service, error) {
+func createBetaCloudService(ctx context.Context, vendorVersion string, tokenSource oauth2.TokenSource, computeEndpoint string) (*computebeta.Service, error) {
 	client, err := newOauthClient(ctx, tokenSource)
 	if err != nil {
 		return nil, err
 	}
-	service, err := computebeta.NewService(ctx, option.WithHTTPClient(client))
+
+	computeOpts := []option.ClientOption{option.WithHTTPClient(client)}
+	if computeEndpoint != "" {
+		computeOpts = append(computeOpts, option.WithEndpoint(computeEndpoint))
```

Comment:
Sorry if I've missed convo on this: does it make sense to have a validation on the endpoint passed in?

Reply:
What kind of validation did you have in mind? For GKE this endpoint will be passed in from a trusted place.